gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -841,37 +841,40 @@ function TellTraitorsAboutTraitors()
 		end
 	end
 
-	hook.Run("TTT2TellTraitors", traitornicks)
+	for _, v in ipairs(player.GetAll()) do
+		if v:HasTeam(TEAM_TRAITOR) then
 
-	-- This is ugly as hell, but it's kinda nice to filter out the names of the
-	-- traitors themselves in the messages to them
-	local traitornicks_min = #traitornicks < 2
+			local tmp = table.Copy(traitornicks)
 
-	if not traitornicks_min then
-		table.sort(traitornicks, function(a, b)
-			return a and b and a:upper() < b:upper()
-		end)
-	end
+			local shouldShow
 
-	for _, v in ipairs(player.GetAll()) do
-		if v:HasTeam(TEAM_TRAITOR) then
-			if traitornicks_min then
-				LANG.Msg(v, "round_traitors_one")
+			shouldShow = hook.Run("TTT2TellTraitors", tmp, v)
+
+			if shouldShow == false or tmp == nil then continue end
```

Comment:
add `or #tmp == 0`. Wouldn't make sense to keep process if the table is empty

Reply:
Well sure, if it's empty, it will show the "Traitor you stand alone" message.

Reply:
Also about `table.RemoveByValue(tmp, v:Nick())`, how are you doing it? Then I can change it

Reply:
If it is empty, you aren't in the list too. So this shouldn't be displayed for you, right? Focused on the roles which manipulate roles like the spy. For sure, the shouldShow method should be used instead, but if you aren't on the list, this shouldn't be shown too...

Reply:
> If it is empty, you aren't in the list too. So this shouldn't be displayed for you, right? Focused on the roles which manipulate roles like the spy. For sure, the shouldShow method should be used instead, but if you aren't on the list, this shouldn't be shown too...

Yeah I understand that. I also had a little "miss-tought". Will add the check.

Reply:
https://github.com/TTT-2/TTT2/pull/361/commits/b8119c88b2a95c9a3a4603c49d4e39dc698233fd
maybe now